src/schema-generation/flex-search-generator.ts

```diff
@@ -0,0 +1,186 @@
+            return filterNode;
+        } else {
+            return simplifyBooleans(new BinaryOperationQueryNode(filterNode, BinaryOperator.AND, searchFilterNode));
+        }
+
+    }
+
+    private getPreExecQueryNode(rootEntityType: RootEntityType, args: { [p: string]: any }, context: QueryNodeResolveInfo): QueryNode {
+        const itemVariable = new VariableQueryNode(decapitalize(rootEntityType.name));
+        const flexSearchType = this.flexSearchTypeGenerator.generate(rootEntityType, false);
+        const flexFilterNode = this.buildFlexSearchFilterNode(args, flexSearchType, itemVariable, rootEntityType, context);
+        return new BinaryOperationQueryNode(
+            new CountQueryNode(
+                new FlexSearchQueryNode({
+                    rootEntityType: rootEntityType,
+                    flexFilterNode: flexFilterNode,
+                    itemVariable: itemVariable
+                })),
+            BinaryOperator.GREATER_THAN,
+            new LiteralQueryNode(context.flexSearchMaxFilterableAmountOverride ? context.flexSearchMaxFilterableAmountOverride : DEFAULT_ARANGOSEARCH_MAX_FILTERABLE_AMOUNT));
+    }
+
+    private augmentWithCondition(schemaField: QueryNodeField, rootEntityType: RootEntityType): QueryNodeField {
+        return {
+            ...schemaField,
+            transform: (sourceNode, args, context) => {
+                const assertionVariable = new VariableQueryNode();
+                if (args[FILTER_ARG] || args[ORDER_BY_ARG]) {
+                    return new WithPreExecutionQueryNode({
+                        preExecQueries: [
+                            new PreExecQueryParms({ resultVariable: assertionVariable, query: this.getPreExecQueryNode(rootEntityType, args, context) })
+                        ],
+                        resultNode: new ConditionalQueryNode(
+                            assertionVariable,
+                            new RuntimeErrorQueryNode(TOO_MANY_OBJECTS_ERROR, { code: FLEX_SEARCH_TOO_MANY_OBJECTS }),
+                            sourceNode)
+                    });
+                } else {
+                    return sourceNode;
+                }
+
+            }
+        };
+    }
+
+    private buildFlexSearchSearchFilterNode(rootEntityType: RootEntityType, itemVariable: VariableQueryNode, expression?: string): QueryNode {
+        if (!expression || expression == '') {
+            return new ConstBoolQueryNode(true);
+        }
+
+        function getQueryNodeFromField(field: Field, path: Field[] = []): QueryNode {
+            if (field.type.isObjectType) {
+                return field.type.fields.map(value => getQueryNodeFromField(value, path.concat(field))).reduce(or, ConstBoolQueryNode.FALSE);
+            }
+
+            function getIdentityNode() {
+                if (field.type.isScalarType && (field.type.graphQLScalarType.name === 'Int' || field.type.graphQLScalarType.name === 'Float') && !isNaN(Number(expression))) {
```

Comment:
to avoid magic strings, you can use `GraphQLInt.name`

Reply:
also, `NaN` seems to go to the starts-with branch, probably not intended? Looks strange. `NaN` should never occur because the `Int` and `Float` types do not allow it, so you could just remove the condition, but if you want to leave it in, you shouldn't skip to the starts-with branch.

Reply:
@Yogu I changed it to the following to make it more clear:
```
                if (field.type.isScalarType && (field.type.graphQLScalarType.name === GraphQLInt.name || field.type.graphQLScalarType.name === GraphQLFloat.name)) {
                    // if the field and the expression are numbers, compare with equals
                    if (!isNaN(Number(expression))) {
                        return new BinaryOperationQueryNode(new FieldPathQueryNode(itemVariable, path.concat(field)), BinaryOperator.EQUAL, new LiteralQueryNode(Number(expression)));
                    } else {
                        // if the field is a number but the expression is not, don't search for this field
                        return new ConstBoolQueryNode(false);
                    }
                } else {
                    // if the field is not a number, compare with starts_with
                    return new FlexSearchStartsWithQueryNode(new FieldPathQueryNode(itemVariable, path.concat(field)), new LiteralQueryNode(expression));
                }
```
The code checks if both the field, and the expression are numbers. 
With this change it also no longer incorrectly searches number-fields with starts_with, if the expression is not a number.

Is there maybe a better way than this, to check if the expression is a number?
`!isNaN(Number(expression))`

Reply:
`isNaN` does not actually check if something is a number (or not a number), it checks if a value is `NaN`. If you pass something else than a number, it converts it to a number first.

What exactly do you want to check here? Is this just defensive programming, or do you want to catch a specific case that could occur under normal circumstances? If you just want to be safe, it might be better to `throw` instead of silently ignoring the filter.